rfcs/replace_ssh_with_ssm_for_linux_template/replace_ssh_with_ssm_for_linux_template.md

```diff
@@ -0,0 +1,238 @@
+    Value: rdp
+```
+
+The step 8 and step 9 are the main logic that user connect workspace resources. Following section will show more detail.
+
+## Design of linux EC2 instance connection
+### UI experience of linux ec2 instance connection
+Let's know about current user experience of linux ec2 instance connection at first:
+![swb_connect_ec2_ssh_ui](./swb_connect_ec2_ssh_ui.png)
+
+1. User clicks "Connections" button, Web UI will show "Create Key" and "Use this SSH Key" if there is existing SSH Key.
+2. User choose a SSH Key, this SSH public key will be sent to EC2 instance, at the same time, UI will show the information to guide user how to access ec2 instance.
+3. User can set CIDRs to limit user to access EC2 instance.
+
+### Linux Workspace Connections Connection Sequence
+Current sequence of linux ec2 instance connection:
+![ssh-design.drawio](./ssh-design.drawio.png)
+
+1. User click "Connections" button, client side will send request to server side to get instanceId, connectionId and connection method. Web UI will render the page to match with connection method. In this case, the connection method is "ssh", Web UI will show the page of "ssh" to user according to "ssh" connection method.
+2. User choose a key which should be created by user before
+3. User click "connect", client side send request to backend to send public key to ec2 instance, then return all connection to client side, suce as "publicDnsName".
+4. Web UI will show the information to guide user how to access ec2 instance.
+5. User ssh to ec2 instance.
+
+# Design Summary
+The general approach proposed by this RFC is replacing ssh connection with AWS System Manager Session Manager. User can access ec2 instance through Session Manager, do not need to store and maintain ssh private key.
+
+## Web UI experience.
+Below is the experience after refactoring. It is similar with sagemaker workspace as below:
+![ssm-ui-experience](./ssm-ui-experience.png)
+
+1. User click "Connections" button, Web UI will show "Connect" button to user.
+2. User click "Connect" button, user will be redirect to AWS Session Manager page, and user can use ec2 instance directly in this page.
+3. There is not "Edit CIDRs" button, because we use swb permission to control which user can access to ec2.
+
+
+## Detail Design.
+For implementing above user experience, there are three parts need to change:
+
+### Linux workspace template
+We need to refactor ``service-workbench-on-aws-cn/addons/addon-base-raas/packages/base-raas-cfn-templates/src/templates/service-catalog/ec2-linux-instance.cfn.yml`` template to support ``Https`` connection method. 
+
+1. As mentioned in previous section, connection method is configured in cloudformation tempalte output value, we need to change linux template as below:
+```
+  MetaConnection1Scheme:
+    Description: Protocol for connection
+    Value: ssh
+```
+===>
+https connection:
+```
+  MetaConnectionScheme:
+    Description: Protocol for connection 1
+    Value: https
```

Comment:
Yes, it will be compatibility with existing SageMaker HTTPS connection. We will have logic to generate different HTTPS url for EC2 session manager and sagemaker notebook instance.

Reply:
For HTTPS connection, does the SWB send post request to backend to handle with the request when clicking the connection?

Reply:
whatever HTTPS or SSH, click green "connections" button, SWB client side send get request to server side to get all this workspace connections information. The web UI will show page to user according to connections information.  For HTTPs connections, SWB send post request to server side to get signin url when user click "Connect" blue button.

Reply:
How will the backend API distinguish the requesting of url connection for SageMaker workspace type and EC2 Linux?

Reply:
Backend API can distinguish the request through "MetaConnectionType" value from workspace template cloudformation output. For Linux, it is "ec2-linux", for SageMaker, it is "SageMaker ".

Reply:
I just concern `https` causing confusing. Is it ok for introducing new `MetaConnectionType`, such as SSM?

Reply:
According to our discussion offline, I will add new connection method just for ssm. Already refactor the rfc doc